### ISSUE TITLE
How quickly "should" an app start?

### ISSUE THREAD

**Issue by chlebowa** [insightsengineering/teal#1234]
### What is your question?

An app built with the following `data` and `modules` (no filters) takes over a minute to start*. The data is not particularly large but larger than the typical example data used in NEST packages. Pre-processing takes about 10 seconds. Since modules do not start until viewed, the (near) entirety of app startup taken up by `teal` itself. I suppose building the filter panel is one of the more time-consuming steps.

*) Measuring in a fresh session, from runnig the script that attaches all necessary packages, creates data and modules, and runs the app, until the moment the notification `Data loaded - App fully started up` appears on screen. 

My question is: at what point should one consider the app taking too long to start? Is there a way to improve startup time?

<details> <summary> 10 datasets of ~30 MB in total and 13 modules </summary>
```
> data
✅︎ verified teal_data object
<environment: 0x55a7eec656c0> [L]
Parent: <environment: package:teal.widgets>
Bindings:
• ADRS: <tibble[,49]> [L]
• ADEX: <tibble[,74]> [L]
• ADLB: <tibble[,103]> [L]
• ADAE: <tibble[,69]> [L]
• ADVS: <tibble[,75]> [L]
• ADSL: <tibble[,39]> [L]
• ADDV: <tibble[,62]> [L]
• ADCM: <tibble[,95]> [L]
• ADTTE: <tibble[,50]> [L]
• ADMH: <tibble[,75]> [L]
```

```
> sapply(ls(get_env(data)), \(x) dim(get_var(data, x))) 
     ADAE ADCM ADDV ADEX ADLB  ADMH ADRS ADSL ADTTE ADVS
[1,] 1934 9216   11  773 8400 24399 1200  400  1600 8982
[2,]   69   95   62   74  103    75   49   39    50   75
```

```
> modules
+ root
 + Summary Tables
  + Data Table
  + Cross Table
 + Data Quality
  + Variable Browser
  + Missing Data
  + Outliers
 + Exploratory Plots
  + Count Barchart
  + Distribution Plot
  + Bivariate Plot
  + Scatter Plot
  + Scatterplot Matrix
 + Statistical Analyses
  + Regression
  + Response Plot
  + Principal Component Analysis
```

</details>

Tagging @gogonzo for visibility.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct.

### Contribution Guidelines

- [X] I agree to follow this project's Contribution Guidelines.

### Security Policy

- [X] I agree to follow this project's Security Policy.

**Comment by chlebowa** [comment]
As evidenced by #1236, the creating the filter panel (`FilteredData` objects) is _not_ the main culprit. Placing progress bars within filter panel ui or server functions did not yield additional information.

**Comment by gogonzo** [comment]
This can be closed. We have implemented following:
- for each module we create a single filteredData object instead of one for every module
- we postpone calling filter panel, transform module and module to the moment when tab is clicked

**Comment by gogonzo** [comment]
> You mean a singleton instead of one instance per module?

@chlebowa Yup
